src/implementations/twig/components/file/file-translation-item.html.twig

```diff
@@ -46,7 +53,9 @@
       }),
       extra_classes: 'ecl-file__translation-download',
       extra_attributes: [
+        { name: 'id', value: _link_id },
         { name: 'download' },
+        { name: 'aria-labelledby', value: '#' ~ _link_id ~ '  #' ~ _title_id ~ ' #' ~  _lang_id },
```

Comment:
aria-labelledby should use the id directly, without the "#"

also, here the link is refereeing to itself (link_id). I think that it would result in the information being said twice. It would be better to refer to the label link only instead.
If not possible (or too complex), we may try to leave the link as is, and add aria-describedby with title_id and lang_id. It should read the link first, then the additional information

Reply:
mmh..i see that in the https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby docs there is mention of the id hash, but why, i wonder..anyway i removed them.

I've created a generated id for the label of the link when it contains an icon and i'm using that in the aria-labelledby of the translation items. Good enough?